config/device-classes/generic/procurve.yaml

```
@@ -0,0 +1,47 @@
+name: procurve
+
+config:
+  components:
+    cpu: true
+    memory: true
+
+match:
+  logical_operator: "OR"
+  conditions:
+    - type: SysObjectID
+      match_mode: startsWith
+      values:
+        - ".1.3.6.1.4.1.11.2.3.7.11"
+
+identify:
+  properties:
+    vendor:
+      - detection: constant
+        value: "HP ProCurve"
+    model:
+      - detection: SysObjectID
+        operators:
+          - type: modify
+            modify_method: map
+            mappings: sysObjectID.yaml
```

Determine the model by the SysDescription so that we don't need to add all the different SysObjectIDs to the mapping and we can read the model of all ProCurve models.
```suggestion
      - detection: SysDescription
        operators:
          - type: modify
            modify_method: regexSubmatch
            regex: '^HP ([^,]+),'
            format: "$1"
```